node/src/types/block.rs

```diff
@@ -2526,141 +2540,33 @@ mod tests {
-
-        fn next(&mut self) -> Option<Self::Item> {
-            let (is_switch_block, validators) = match &self.switch_block_indices {
-                Some(switch_block_indices)
-                    if switch_block_indices.contains(&self.block.height()) =>
-                {
-                    let secret_keys: Vec<SecretKey> = iter::repeat_with(|| {
-                        SecretKey::ed25519_from_bytes(
-                            self.rng.gen::<[u8; SecretKey::ED25519_LENGTH]>(),
-                        )
-                        .unwrap()
-                    })
-                    .take(4)
-                    .collect();
-                    let validators: BTreeMap<_, _> = secret_keys
-                        .iter()
-                        .map(|sk| (PublicKey::from(sk), 100.into()))
-                        .collect();
-
-                    (true, Some(validators))
-                }
-                Some(_) | None => (false, None),
-            };
-
-            let next = Block::new(
-                *self.block.hash(),
-                self.block.header().accumulated_seed(),
-                *self.block.header().state_root_hash(),
-                FinalizedBlock::random_with_specifics(
-                    &mut self.rng,
-                    self.block.header().era_id(),
-                    self.block.header().height() + 1,
-                    is_switch_block,
-                    Timestamp::now(),
-                    iter::empty(),
-                ),
-                validators,
-                self.block.header().protocol_version(),
-            )
-            .unwrap();
-            self.block = next.clone();
-            Some(next)
-        }
-    }
-
-    #[test]
-    fn test_block_iter() {
-        let rng = TestRng::new();
-        let test_block = TestBlockSpec::new(rng, None);
-        let mut block_batch = test_block.into_iter().take(100);
-        let mut parent_block: Block = block_batch.next().unwrap();
-        for current_block in block_batch {
-            assert_eq!(
-                current_block.header().height(),
-                parent_block.header().height() + 1,
-                "height should grow monotonically"
-            );
-            assert_eq!(
-                current_block.header().parent_hash(),
-                parent_block.hash(),
-                "block's parent should point at previous block"
-            );
-            parent_block = current_block;
-        }
-    }
-
-    #[test]
-    fn test_block_iter_creates_switch_blocks() {
-        let switch_block_indices = vec![0, 10, 76];
-
-        let rng = TestRng::new();
-        let test_block = TestBlockSpec::new(rng, Some(switch_block_indices.clone()));
-        let block_batch: Vec<_> = test_block.into_iter().take(100).collect();
-
-        let base_height = block_batch.first().expect("should have block").height();
+#[cfg(any(feature = "testing", test))]
+pub fn random_era_report(rng: &mut TestRng) -> EraReport {
```

Comment:
This doesn't need to be `pub` I think, but if we remove `random_switch_block` (or implement it using `Block::random_with_specifics`) then this function can also be moved back to inside `FinalizedBlock::random_with_specifics`.

Reply:
Updated in 00bf59f5a353591de2ec1c7346075c02d17df4c4